### PR TITLE
Version 3.1

### PR DESCRIPTION
**Pull request by scomari**
**NEW FEATURES:**
- _Joints PVT control app_:  allows the user to run multiple joints PVTs after homing procedure. Comes together with a user friendly windows with trajectory plots.
- _Support app for external motion capture system calibration_: allows the user to move the platform in freedrive mode, lock it and run an excitation trajectory while recording actuators status. It is currently the only calibration app present.
- _Freedrive mode_: allows the user to freely move the platform controlled in torque control mode. This option can be found in the main GUI
- _Debug app/mode_: allows the user to enable a debug flag that makes a debug button connected to a debug app available. The debug app can be extended by the user depending on the particular need.

**MISC:**
- Bugs fix
- Code documentation extended
- Bumped up grab_common libs with under-actuated functionalities
- Included matlab code as sub-sub-module (in libcdpr)

### CODE REVIEW

File: matlab_updated/CalcGsJacobians.m [hunk 1]
```diff
@@ -1,42 +1,72 @@
...
-      cdpr_v.cable(i).vers_n*dphidq(i,:);
-  dJTp(:,i,:) = drhodq;
-  dadql = [zeros(3,3) -Anti(cdpr_v.cable(i).pos_PA_glob)*cdpr_v.platform.H_mat];
-  dJTo(:,i,:) = Anti(cdpr_v.cable(i).pos_PA_glob)*drhodq-Anti(cdpr_v.cable(i).vers_rho)*dadql;
+  dldq(1,:) = cdpr_v.cable(i).vers_rho'*dadq;
+  drhodq = sin(cdpr_v.cable(i).tan_ang).*cdpr_v.cable(i).vers_w*dsdq(1,:)+...
+      cdpr_v.cable(i).vers_n*dphidq(1,:);
+  dadq(1:3,1:3) = zeros(3);
+  dJT = dJT+[drhodq;Anti(cdpr_v.cable(i).pos_PA_glob)*drhodq-Anti(cdpr_v.cable(i).vers_rho)*dadq].*cdpr_v.tension_vector(i);
+  J_sl(i,:) = dsdq;
```
Comment (scomari) [inline matlab_updated/CalcGsJacobians.m]: La matrice `J_sl` a cosa serve? Di conseguenza, possiamo eliminare anche `dldq`, che mi sembra appaia solo qui?

File: matlab_updated/CalcGsJacobians.m [hunk 2]
```diff
@@ -1,42 +1,72 @@
...
-  drhodq = sin(cdpr_v.cable(i).tan_ang).*cdpr_v.cable(i).vers_w*dsdq(i,:)+...
-      cdpr_v.cable(i).vers_n*dphidq(i,:);
-  dJTp(:,i,:) = drhodq;
-  dadql = [zeros(3,3) -Anti(cdpr_v.cable(i).pos_PA_glob)*cdpr_v.platform.H_mat];
-  dJTo(:,i,:) = Anti(cdpr_v.cable(i).pos_PA_glob)*drhodq-Anti(cdpr_v.cable(i).vers_rho)*dadql;
+  dldq(1,:) = cdpr_v.cable(i).vers_rho'*dadq;
+  drhodq = sin(cdpr_v.cable(i).tan_ang).*cdpr_v.cable(i).vers_w*dsdq(1,:)+...
+      cdpr_v.cable(i).vers_n*dphidq(1,:);
+  dadq(1:3,1:3) = zeros(3);
+  dJT = dJT+[drhodq;Anti(cdpr_v.cable(i).pos_PA_glob)*drhodq-Anti(cdpr_v.cable(i).vers_rho)*dadq].*cdpr_v.tension_vector(i);
```
Comment (scomari) [inline matlab_updated/CalcGsJacobians.m]: `Anti(cdpr_v.cable(i).pos_PA_glob)*drhodq-Anti(cdpr_v.cable(i).vers_rho)*dadq`
Questa matrice è sempre 3xm, ma non dovrebbe essere (m-3)xm invece?
Funziona ovviamente per il caso di SO(3), ma non in quello (seppur futuro) in cui usiamo i quaternioni, no?

File: matlab_updated/CalcGsJacobians.m [hunk 3]
```diff
@@ -1,42 +1,72 @@
-function J_q = CalcGsJacobians(cdpr_v,Ja,Ju,mg)
+function J_opt = CalcGsJacobians(cdpr_v,Ja,Ju,mg)
+
+% n = length(Ja);
```
Comment (scomari) [inline matlab_updated/CalcGsJacobians.m]: Non serve commentare il codice vecchio.. Le versioni servono a questo! ;)

**Review by scomari — COMMENTED** [review]
Vedi commenti

**Review by EdoardoIda — APPROVED** [review]

**Pull request merged**